### DISCUSSION TITLE
Restoring Backup

### DISCUSSION THREAD

**Discussion by Rom0067** [mbround18/valheim-docker#850] [Help!] [answered]
I am sorry if this is a dumb question but I searched around and couldn't find a direct answer.

I am running a container on a Linux machine and everything is working fine including backups are being created.

If something happens and I need to restore from one of these said back ups how do I do it from docker compose? Do I need to change something in the container files?  Again I'm sorry I'm pretty new to this stuff.  Thank you kindly for any help.

**Comment by mbround18** [comment]
My apologies I'm remodeling so I didn't have a chance to see this inquiry, your vest bet is to un tar. You can exec into the container and 

tar -xvzf /path/of/backup -C /path/to/save/directory

**Reply by Rom0067** [reply]
Thank you for replying with your busy schedule!  

Again I am new to self hosting in general so if I might ask a couple questions.

1. Do I need to delete or move everything in my current save file and replace it with the contents of the back up tar file.
2.If that is the case do I then need to compose down the container && compose up for the back up to be read?
3.  Does the code that you posted extract all the files within the tar to the save directory?
4. Optional question you don't have to answer I'll continue my research:  Im unable to exec in via your example (docker exec -it $CONTAINER_NAME gosu steam bash) I get gosu permission errors.  I was able to exec If I sudo su and remove gosu. I apologize I'm sure this is a newb question and may be how I'm supposed to do it.

Thank you very much again for taking the time to help me!

**Reply by mbround18** [reply, marked as answer]
If you are doing a saverestore yes, remove the contents and extract. If you are unsure you can back up the original contents before atteampting a restore.

yes as long as you change the place holder paths 

we changed the container, i need to update the docs,  its now `docker exec -it $CONTAINER_NAME bash` the container runs as user not as root which was a security risk

**Reply by Rom0067** [reply]
Awesome Thanks again!

1. So process should be:  back up save folder, delete contents of folder, use tar -xvzf /path/of/backup -C /path/to/save/directory, reboot server.  That's easy enough!
2. Thanks got it!
3. That explains why I was having issues. Especially being new I had no idea what was wrong.  I just removed gosu and  sudo su and it worked thought I was a genius or something lol.  Just tried what you said and it worked!

Overall thank you for answering all my questions I think I'm in a good place for now.  Thanks for all you do!

(Good luck on the remodel.  Those can be a pain!)
